// discloud.app/dist/structures/App.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const BaseApp_1 = __importDefault(require("./BaseApp"));
class App extends BaseApp_1.default {
    /**
     * If your app has auto deploy on github enabled
     */
    autoDeployGit;
    /**
     * Your app's exit code on stopping
     */
    exitCode;
    /**
     * Moderators IDs of your app
     */
    mods;
    /**
     * If your app is online
     */
    online;
    /**
     * If your application was stopped due to lack of RAM
     */
    ramKilled;
    constructor(discloudApp, data) {
        super(discloudApp, data);
        this.autoDeployGit = data.autoDeployGit;
        this.exitCode = data.exitCode;
        this.mods = data.mods ?? [];
        this.online = data.online ?? false;
        this.ramKilled = data.ramKilled;
    }
    async restart() {
        const data = await super.restart();
        this.online = data.status === "ok";
        return data;
    }
    async start() {
        const data = await super.start();
        this.online = data.status === "ok";
        return data;
    }
    async stop() {
        const data = await super.stop();
        this.online = !(data.status === "ok");
        return data;
    }
    async update(options) {
        const data = await super.update(options);
        this.online = data.statusCode === 200;
        return data;
    }
}
exports.default = App;
