// discloud.app/dist/structures/AppStatus.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const util_1 = require("@discloudapp/util");
const Base_1 = __importDefault(require("./Base"));
class AppStatus extends Base_1.default {
    /**
     * Status of your application
     * - It can be `Online` or `Offline` for example
     */
    container;
    /**
     * CPU usage as percentage
     */
    cpu;
    /**
     * Your app id
     */
    id;
    /**
     * Relative time of the last restart
     */
    lastRestart;
    /**
     * RAM usage
     */
    memory;
    /**
     * Percentage of memory usage
     */
    memoryUsage;
    /**
     * Internet usage
     */
    netIO;
    /**
     * Storage space
     */
    ssd;
    /**
     * Date of your application has started
     */
    startedAt;
    /**
     * Timestamp of when your app started
     */
    startedAtTimestamp;
    constructor(discloudApp, data) {
        super(discloudApp);
        this.container = data.container;
        this.cpu = data.cpu;
        this.id = data.id;
        this.lastRestart = data.last_restart;
        if (data.memory) {
            this.memory = data.memory;
            const matched = data.memory.match(/[\d.]+/g) ?? [];
            this.memoryUsage = (0, util_1.calculatePercentage)(matched[0], matched[1]);
        }
        this.netIO = data.netIO;
        this.ssd = data.ssd;
        if (data.startedAt) {
            this.startedAt = new Date(data.startedAt);
            this.startedAtTimestamp = this.startedAt.valueOf();
        }
    }
}
exports.default = AppStatus;
